backend/dal/Services/Concrete/ParcelService.cs

```diff
@@ -341,11 +348,14 @@ public void Remove(Parcel parcel)
                 throw new NotAuthorizedException("User does not have permission to delete.");
 
             this.Context.Entry(entity).CurrentValues.SetValues(parcel);
+            this.User.CanUpdateOrRemoveParcelOrThrow(this.Context, entity);
+
             entity.Buildings.ForEach((building) =>
             {
                 this.Context.BuildingEvaluations.RemoveRange(building.Evaluations);
                 this.Context.BuildingFiscals.RemoveRange(building.Fiscals);
                 this.Context.Buildings.Remove(building);
+                this.User.CanUpdateOrRemoveBuildingOrThrow(this.Context, building);
```

Comment:
Ideally you would do this before performing the above statements as it short-circuits with it.